set-web/src/main/java/eu/europeana/set/web/service/UserSetService.java

```diff
@@ -220,4 +222,7 @@ public ItemIdsResultPage buildItemIdsResultsPage(List<String> itemIds, int page,
      * @return
      */
     List<PersistentUserSet> getEntitySetBestBetsItems(UserSetQuery query);
+    
+    void checkDuplicateUserSets(UserSet userSet, boolean withoutItself) throws SetUniquenessValidationException;
```

Comment:
the withoutItself parameter is redundant. The method should simply check if the identifier exists in the provided userset.

Reply:
please remove the boolean flag from all methods ...

Reply:
I do not think it is redundant. For the update method, if some other field (other then the subject that is used for the duplication check) is changed, if there is no slef id check, the duplicate check will return the self id, becuase the subject is not changed, and in that way throw an exception and prevent updating the set.